### PR TITLE
Schema check for mapTreeFromNodeData

### PR DESCRIPTION
**Pull request by daesunp**
## Description

This PR adds a schema check when getting the mapTree from node data.

### CODE REVIEW

File: packages/dds/tree/src/simple-tree/toMapTree.ts [hunk 1]
```diff
@@ -188,6 +189,9 @@ function nodeDataToMapTree(
 	// They already have the mapTree, so there is no need to recompute it.
 	const flexNode = tryGetFlexNode(data);
 	if (flexNode !== undefined) {
+		if (!allowedTypes.has(tryGetSimpleNodeSchema(flexNode.schema) ?? fail("missing schema"))) {
```
Comment (noencke) [inline packages/dds/tree/src/simple-tree/toMapTree.ts:192]: We should probably put this error inside of the `isMapTreeNode` check. That means that if a user tries to insert a tree that is already inserted _and_ is out of schema, they will get the "already inserted" error rather than the "out of schema" error. I think the former is more fundamental because it indicates the user is doing something categorically wrong (re-inserting trees is always illegal, everywhere) rather than merely wrong in this context (the schema doesn't match up here, but it probably does somewhere else).

File: packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts [hunk 2]
```diff
@@ -540,19 +541,36 @@ describe("toMapTree", () => {
 
 		it("Throws on schema-incompatible entries", () => {
 			const schemaFactory = new SchemaFactory("test");
-			const schema = schemaFactory.map("map", schemaFactory.string);
+			const schema = schemaFactory.object("testObject", {
+				field1: schemaFactory.string,
```
Comment (noencke) [inline packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts:545]: What is the reason for this change?

Comment (daesunp) [reply packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts]: Whoops, was checking something and forgot to change it back. Reverted the change now :)

File: packages/dds/tree/src/simple-tree/toMapTree.ts (hunk 1 above)
Comment (daesunp) [reply packages/dds/tree/src/simple-tree/toMapTree.ts]: Moved usage error to inside of the `isMapTreeNode` check

**Review by noencke — APPROVED** [review]
